index.bs

```diff
@@ -182,21 +193,55 @@ To <dfn>Construct a OrientationSensor Object</dfn> the user agent must invoke th
 The {{OrientationSensor/populateQuaternion()}} method must run these steps or their [=equivalent=]:
     1.  If |targetBuffer| is not of type {{Float32Array}}, [=throw=] a
         "{{SyntaxError!!exception}}" {{DOMException}} and abort these steps.
-    1.  If |targetBuffer| is of type {{Float32Array}} with a size other than four, [=throw=] a
+    1.  If |targetBuffer| is of type {{Float32Array}} with a size less than four, [=throw=] a
         "{{SyntaxError!!exception}}" {{DOMException}} and abort these steps.
     1.  If the value of |sensor_instance|.[=[[state]]=] is not <a enum-value>"activated"</a>,
         [=throw=] an "{{InvalidStateError!!exception}}" {{DOMException}} and abort these steps.
-    1.  Run these sub-steps [=in parallel=]:
+    1.  Run these sub-steps:
         1.  Let |quaternion| be the value of [=latest reading=]["quaternion"]
         1.  If |quaternion| is `null`, abort these steps.
         1.  Copy |quaternion| into |targetBuffer|
 </div>
 
 ### OrientationSensor.populateMatrix() ### {#orientationsensor-populatematrix}
 
-<div algorithm="to populate matrix">
-The {{OrientationSensor/populateMatrix()}} method must run these steps or their [=equivalent=]:
+The {{OrientationSensor/populateMatrix()}} method  populates the given buffer with WebGL-compatible rotation matrix
+which is converted from the value of [=latest reading=]["quaternion"] [[QUATCONV]], as shown below
 
+<img src="quaternion_to_rotation_matrix.png" style="display: block;margin: auto;" alt="Converting quaternion to rotation matrix." width="50%" height="50%">
+
+where:
+
+ - W = cos(θ/2).
+ - X = V<sub>x</sub> * sin(θ/2).
+ - Y = V<sub>y</sub> * sin(θ/2).
+ - Z = V<sub>z</sub> * sin(θ/2).
+
+<div algorithm="to populate WebGL-compatible matrix">
+The {{OrientationSensor/populateMatrix()}} method must run these steps or their [=equivalent=]:
+    1.  If |targetBuffer| is not of type {{Float32Array}}, [=throw=] a
+        "{{SyntaxError!!exception}}" {{DOMException}} and abort these steps.
+    1.  If |targetBuffer| is of type {{Float32Array}} with a size less than sixteen, [=throw=] a
+        "{{SyntaxError!!exception}}" {{DOMException}} and abort these steps.
```

Comment:
I would propose to use TypeError. https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror

> TypeError is used to indicate an unsuccessful operation when none of the other NativeError objects are an appropriate indication of the failure cause.

wdyt?